packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1389,6 +1390,12 @@ function updateSuspenseComponent(
   const mode = workInProgress.mode;
   const nextProps = workInProgress.pendingProps;
 
+  if (__DEV__) {
+    if (shouldSuspend(workInProgress)) {
+      workInProgress.effectTag |= DidCapture;
```

Comment:
Suggestion: replace this with

```js
throwException(
  root,
  workInProgress, // returnFiber
  null, // sourceFiber
  dummyPromiseThatNeverResolves,
  renderExpirationTime
);
```

This will require refactoring `throwException` a bit, or perhaps splitting it into multiple functions, since it currently assumes that you have a source fiber.